src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2799,6 +2907,20 @@ public override SyntaxNode WithStatements(SyntaxNode declaration, IEnumerable<Sy
             }
         }
 
+        private SyntaxNode FixDeclarationWithStatements(SyntaxNode declaration)
+        {
+            if (HasBody(declaration))
+            {
+                var mods = GetModifiers(declaration);
+                if (mods.IsAbstract)
+                {
+                    return WithModifiers(declaration, mods.WithIsAbstract(false).WithIsVirtual(true));
```

Comment:
CodeModel is hte *last* API i would personally use when judging if a decision is ok or not :D